include/kernel_version.h

```diff
@@ -30,6 +30,62 @@ extern "C" {
 #define SYS_KERNEL_VER_MINOR(ver) (((ver) >> 16) & 0xFF)
 #define SYS_KERNEL_VER_PATCHLEVEL(ver) (((ver) >> 8) & 0xFF)
 
+struct version_header {
+	/** Always equal to $B!N */
+	u8_t magic[4];
```

Comment:
Why this is required? I expect that address of this structure instance will be obtained from the elf or map file.

Reply:
It is to uniquely identify the version_header structure.